Add Tableau role and group Ids field to Invitation model

Jtang-1 opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
No user facing effects

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4450)

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
This change in itself is not specific to a feature flag but associated feature change will be specific to [Tableau User Syncing
](https://www.commcarehq.org/hq/flags/edit/tableau_user_syncing/)

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
This change only adds a field to an existing model. The field will be null if not populated.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
tests exists that Invitations can be created and accepted/declined as expected but there is no test yet for this specific field.

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No QA

### Migrations
<!-- Delete this section if the PR does not contain any migrations -->
<!-- https://commcare-hq.readthedocs.io/migrations_in_practice.html -->
- [X] The migrations in this code can be safely applied first independently of the code

<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

Jtang-1 committed:
adds tableau roles and group ids to Invitation

Robert-Costello commented on corehq/apps/users/models.py:
```diff
@@ -2686,6 +2687,8 @@ class Invitation(models.Model):
                                 on_delete=models.SET_NULL, null=True)
     custom_user_data = models.JSONField(default=dict)
+    tableau_role = models.CharField(max_length=32, choices=TABLEAU_ROLES, null=True)
+    tableau_group_ids = ArrayField(models.CharField(max_length=32), null=True)
```
Does that limit  comes from here?
https://github.com/dimagi/commcare-hq/blob/4d72daac80b0d0ae34249f875f75e88e7f80811c/corehq/apps/reports/models.py#L251

Jtang-1 replied:
I saw the `TableauGroup` model that was eventually deleted in this [PR](https://github.com/dimagi/commcare-hq/pull/32379/files) had a 32 char limit so I based it off that

Robert-Costello replied:
I'm a little confused. Are we expecting the array of group ids to be strings like `HQ_TABLEAU_GROUP_NAME = 'HQ'` or some alphanumeric uuid's?

Robert-Costello replied:
If we know it will be id's in this format `4a30894a-dd66-4cbb-b14c-4f701135c9c5` of 36 chars, I would assume it would be safe to assume consistency and make the max 36 chars.

Jtang-1 replied:
Thanks for looking into this Addision, I should have checked myself before setting it to 32. Made the change t 36 here. [a07d982](https://github.com/dimagi/commcare-hq/pull/34445/commits/a07d982d24d511e4d46912fefea526f23172fda3)

Jtang-1 committed:
bug: tableau group ids have 36 characters

We weren't able to find documentation by Tableau stating that it will be 36
characters but examples shown and generated ids have been 36

Jtang-1 commented on corehq/apps/users/migrations/0059_invitations_addtableau_roles_and_groupids.py:
```diff
@@ -0,0 +1,24 @@
+        migrations.AddField(
+            model_name='invitation',
+            name='tableau_role',
+            field=models.CharField(choices=[('Explorer - Can Publish', 'ExplorerCanPublish'), ('Server Administrator', 'ServerAdministrator'), ('Site Administrator - Explorer', 'SiteAdministratorExplorer'), ('Site Administrator - Creator', 'SiteAdministratorCreator'), ('Unlicensed', 'Unlicensed'), ('Read Only', 'ReadOnly'), ('Viewer', 'Viewer')], max_length=32, null=True),
```
That's how I understand it, that it'll require a new migration

Jtang-1 committed:
Merge branch 'master' into jt/invitation-tableau-user

AddisonDunn reviewed (changes requested):
.

Jtang-1 commented on corehq/apps/users/models.py:
```diff
@@ -2686,6 +2687,8 @@ class Invitation(models.Model):
                                 on_delete=models.SET_NULL, null=True)
     custom_user_data = models.JSONField(default=dict)
+    tableau_role = models.CharField(max_length=32, choices=TABLEAU_ROLES, null=True)
+    tableau_group_ids = ArrayField(models.CharField(max_length=36), null=True)
```
Would we be able the get the group name from Tableau via the group id? My understanding is that Tableau is the source of truth for available groups

Jtang-1 replied:
Another point on that, I imagine the group name could also change? So if we store the group name, I think there could potentially be a stale data issue where the group name changes between the creation of the invite and the creation of the web user. In that case, the `allowed_groups_for_domain` check would be incorrect.

Jtang-1 replied:
Appreciate your insight and it makes sense. 

I'm imagining manipulation will need to be done anyways to create `TableauGroupTuples` before using `update_tableau_user` so the group names would be retrieved then. I'm leaning towards doing that since storing the group name in the `Invitation` model feels redundant and my preference is to stick closer to the "Tableau is the source of truth" framework. So going to proceed with that

Jtang-1 committed:
correction migration file name
